client/state/billing-transactions/individual-transactions/actions.ts

```diff
@@ -0,0 +1,74 @@
+import { translate } from 'i18n-calypso';
+import wp from 'calypso/lib/wp';
+import { billingHistoryReceipt } from 'calypso/me/purchases/paths';
+import {
+	BILLING_TRANSACTION_ERROR_CLEAR,
+	BILLING_TRANSACTION_REQUEST,
+	BILLING_TRANSACTION_RECEIVE,
+	BILLING_TRANSACTION_REQUEST_FAILURE,
+	BILLING_TRANSACTION_REQUEST_SUCCESS,
+} from 'calypso/state/action-types';
+import { errorNotice } from 'calypso/state/notices/actions';
+import type { IndividualReceipt } from '../types';
+import type { CalypsoDispatch } from 'calypso/state/types';
+
+import 'calypso/state/billing-transactions/init';
+
+export const requestBillingTransaction =
+	( transactionId: number | string ) => ( dispatch: CalypsoDispatch ) => {
```

Comment:
I wonder if it makes sense to cast this to a number or a string somewhere, so we're certain that we're working with a single data type. I've seen subtle issues coming from the fact that we expect data to be either of these types but we use it as the other one.

Reply:
I think you're right, but I typed this to match [the PropType on this data's query component](https://github.com/Automattic/wp-calypso/blob/8349c684ccb816603498bf520220674ee50b2a9c/client/components/data/query-billing-transaction/index.jsx#L28). That said, it's only in use in two places, `ReceiptView` and `BillingReceipt`. `ReceiptView` gets it directly from the URL here (which I assume makes it a string):

https://github.com/Automattic/wp-calypso/blob/8349c684ccb816603498bf520220674ee50b2a9c/client/my-sites/purchases/controller.js#L87

And `BillingReceipt` gets it from the URL also:

https://github.com/Automattic/wp-calypso/blob/8349c684ccb816603498bf520220674ee50b2a9c/client/me/purchases/billing-history/controller.js#L11

So I think it's safe to consider it a string. The same is true for `clearBillingTransactionError()`.

The tests treat the ID as a number, though...

https://github.com/Automattic/wp-calypso/blob/8349c684ccb816603498bf520220674ee50b2a9c/client/state/billing-transactions/individual-transactions/test/actions.js#L117

I have a feeling that the proper thing to do is to actually use a number here and do some light casting. I've pushed commits to do that.